craft_cli/messages.py

```diff
@@ -167,14 +167,15 @@ def __init__(self):
         self.initiated = False
         self.log_filepath = None
 
-    def init(self, mode: EmitterMode, greeting: str):
+    def init(self, mode: EmitterMode, appname: str, greeting: str):
```

Comment:
Because `__init__` is called on module loading, when everything is bootstrapped by Python (this will leave a module-level `emit` that is imported and used everywhere), while `init` is called by the application module to really put Emitter to work (creating a log file, etc) while passing the application related information (the default initial mode, the application name, and the greeting).

Reply:
ah, if I got it right, then can we add a docstring to Emitter that it is not meant to be instantiated by a user of the api. There's also a typo in that doctstring "s/This handling everything/This handles everything"

Reply:
Added a line in the docstring for it (and fixed the typo).

Thinking about this, we could even make Emitter private (`_Emitter`). Originally it was public because the "modes" were an attribute of the class, but now it's outside, so...